src/cascadia/PublicTerminalCore/HwndTerminal.cpp

```diff
@@ -324,26 +323,18 @@ void HwndTerminal::_UpdateFont(int newDpi)
 
 IRawElementProviderSimple* HwndTerminal::_GetUiaProvider() noexcept
 {
-    if (nullptr == _uiaProvider && !_uiaProviderInitialized)
+    if (!_uiaProvider)
     {
-        std::unique_lock<std::shared_mutex> lock;
         try
         {
-#pragma warning(suppress : 26441) // The lock is named, this appears to be a false positive
-            lock = _terminal->LockForWriting();
-            if (_uiaProviderInitialized)
-            {
-                return _uiaProvider.Get();
-            }
-
+            auto lock = _terminal->LockForWriting();
             LOG_IF_FAILED(::Microsoft::WRL::MakeAndInitialize<::Microsoft::Terminal::TermControlUiaProvider>(&_uiaProvider, this->GetUiaData(), this));

```

Comment:
I believe we used the boolean `_uiaProviderInitialized` to avoid throwing over and over and over if there was some failure in establishing a UIA Provider. I don't know why there _would_ be however.

Reply:
Let's document this then! :)

Reply:
Do you know why we used to check `_uiaProviderInitialized` again after locking the terminal?

Reply:
Probably somebody trying to do "double-checked locking", in case it changed while outside of lock.

Reply:
Is that still the case? Can this function be called from multiple threads? This would be a race condition...

Reply:
I don't believe so. _GetUiaProvider is only called from the window's message loop, which by virtue of being a WndProc always fires on the window's message pump thread.